File: data-inspector-scala/src/main/scala/com/example/datainspector/model/QueryResult.scala
/*
 * Copyright (c) 2026 Data Inspector Contributors
 * Licensed under the MIT License. See LICENSE file in the project root.
 */
package com.example.datainspector.model

import scala.beans.BeanProperty
import scala.jdk.CollectionConverters._

/**
 * Result of querying a data source
 */
case class QueryResult(
  @BeanProperty dataSourceId: String,
  @BeanProperty data: java.util.List[java.util.Map[String, AnyRef]],
  @BeanProperty totalCount: Long,
  @BeanProperty limit: Int = 100,
  @BeanProperty offset: Int = 0,
  @BeanProperty stats: java.util.Map[String, AnyRef] = java.util.Collections.emptyMap()
)

object QueryResult {
  def apply(dataSourceId: String,
            data: List[Map[String, AnyRef]],
            totalCount: Long,
            limit: Int,
            offset: Int): QueryResult = {
    val javaData = data.map(_.asJava).asJava
    QueryResult(dataSourceId, javaData, totalCount, limit, offset,
      java.util.Collections.emptyMap[String, AnyRef]())
  }

  def apply(dataSourceId: String,
            data: List[Map[String, AnyRef]],
            totalCount: Long,
            limit: Int,
            offset: Int,
            stats: Map[String, AnyRef]): QueryResult = {
    val javaData = data.map(_.asJava).asJava
    QueryResult(dataSourceId, javaData, totalCount, limit, offset, stats.asJava)
  }
}

File: data-inspector/src/main/java/io/github/kanxiao/spring/inspector/model/QueryResult.java
/*
 * Copyright (c) 2026 Data Inspector Contributors
 * Licensed under the MIT License. See LICENSE file in the project root.
 */
package io.github.kanxiao.spring.inspector.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.Map;

/**
 * Result of querying a data source
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class QueryResult {
    private String dataSourceId;
    private List<Map<String, Object>> data;
    private long totalCount;
    private int limit;
    private int offset;
    private Map<String, Object> stats;
}
